### ISSUE TITLE
testID prop is converted to `data-testid` in dev but `testid` in prod

### ISSUE THREAD

**Issue by la55u** [tamagui/tamagui#1368]
### Current Behavior

Sometimes the `testID` prop (`data-testid` attribute) is getting transformed by the compiler (?) to `testid`, which causes a different output in dev vs prod. I tested it with an other `data-random` prop too but that seems to be correctly preserved in all 3 cases in the demo below. 

### Expected Behavior

It should always be `data-testid` since that is the valid HTML.

### Tamagui Version

```markdown
1.36.6
```

### Reproduction

Minimal reproduction: https://github.com/la55u/tama-testid-bug

1. clone the repo, run `yarn && yarn web`
2. confirm all 3 testID props appear as `data-testid` attributes in the DOM
3. run `yarn web:prod && yarn web:prod:serve`
4. you should see that on YStack and Paragraph the `data-testid` became `testid`, but remained unchanged on Button

### System Info

```markdown
System:
    OS: Linux 6.3 Fedora Linux 38 (Workstation Edition)
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-1260P
    Memory: 19.14 GB / 31.04 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 3.5.0 - ~/.nvm/versions/node/v18.12.1/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
    Firefox: 115.0b9
  npmPackages:
    @babel/runtime: ^7.18.9 => 7.21.0 
    @manypkg/cli: ^0.19.1 => 0.19.2 
    check-dependency-version-consistency: ^3.0.3 => 3.0.3 
    eslint: ^8.21.0 => 8.39.0 
    node-gyp: ^9.3.1 => 9.3.1 
    prettier: ^2.7.1 => 2.8.8 
    turbo: ^1.10.3 => 1.10.3 
    typescript: ^5.1.3 => 5.1.3
```

**Comment by natew** [comment]
check latest please

**Comment by la55u** [comment]
fixed in 1.38.2, thank you!
